### PR TITLE
feat(swap): add link to provider terms and conditions

### LINKED ISSUE #21310: Open provider Terms & Conditions

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #21310 

### Summary

This PR adds a link to the swap provider terms and conditions on the Set Spending Cap and Swap Confirmation screens.

https://github.com/user-attachments/assets/fdc022cf-12e3-4e1e-b462-00acac241872

https://github.com/user-attachments/assets/b5f2f439-df0a-4077-ab88-9f312dd8072b

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

### Steps to test

- Open Status
- Go to swap
- Select and enter valid amounts
- Verify Terms and Conditions is working properly

status: ready

### CODE REVIEW

**Review by mohsen-ghafouri — APPROVED** [review]
LGTM

**Review by clauxx — APPROVED** [review]

File: src/status_im/contexts/wallet/swap/set_spending_cap/view.cljs [hunk 1]
```diff
@@ -228,9 +228,24 @@
 
 (defn- footer
   []
-  [rn/view {:style {:margin-bottom -10}}
-   [transaction-details]
-   [slide-button]])
+  (let [provider (rf/sub [:wallet/swap-proposal-provider])
+        theme    (quo.theme/use-theme)
+        on-press (rn/use-callback #(rf/dispatch [:open-url (:terms-and-conditions-url provider)])
```
Comment (smohamedjavid) [inline src/status_im/contexts/wallet/swap/set_spending_cap/view.cljs]: Quick question: We have a provider check to display the full name, but we don't do the same for the T&C URL.

Comment (briansztamfater) [reply src/status_im/contexts/wallet/swap/set_spending_cap/view.cljs]: This may be a problem but only in development, I will add the check 👍

Comment (briansztamfater) [reply src/status_im/contexts/wallet/swap/set_spending_cap/view.cljs]: Updated :)

**Comment by mariia-skrypnyk** [comment]
Hey @briansztamfater !

Thanks for your PR.
I've tested t on both platforms and have a question regarding navigation on Android.

Here are steps 
1) tap on `Privacy Policy`
2) choose Cancel on `Downloading popup`

On my device I see a blink of some search result zoomed. 
Not sure I understand what it is.
Screenshot 2024-09-26 at 14 26 37

You can see it on my video:
https://github.com/user-attachments/assets/a8b28ff1-849a-4290-a5ec-d39922244bf0

It was NOT catched on another device. So this issue happens only on my Samsung A34 (14.0).
On another device I see just a black screen after `cancel`. 

https://github.com/user-attachments/assets/cf610abb-3f57-4a9b-9951-2a1e9f272bf8

Is it a correct result or something else should happen there?
And yes easily can become a followups if we decide to fix/change anything.
Thanks.

**Comment by briansztamfater** [comment]
@mariia-skrypnyk for me it seems like an old website that was previously loaded? Not sure honestly, but anyways, we can't control blink on the Android explorer as it is happening outside of the Status app. I think we can ignore this issue.

**Comment by mariia-skrypnyk** [comment]
Ok @briansztamfater thanks.

This blink is really smth specific.
Can you answer me on this?

> On another device I see just a black screen after cancel.
> Is it a correct result or something else should happen there?

I see your rebase, need to recheck build.
I ran e2e and let you know regarding merge.

**Comment by briansztamfater** [comment]
> Ok @briansztamfater thanks.
> 
> This blink is really smth specific. Can you answer me on this?
> 
> > On another device I see just a black screen after cancel.
> > Is it a correct result or something else should happen there?
> 
> I see your rebase, need to recheck build. I ran e2e and let you know regarding merge.

Black screen is correct if there is no other website previously loaded on the Android browser. This is all happening on the Android browser, so we have no control of that. So behavior seems correct, at least from the Status app side 👍

**Comment by mariia-skrypnyk** [comment]
Thanks @briansztamfater !

PR can be merged!
Great job!

**Pull request merged**